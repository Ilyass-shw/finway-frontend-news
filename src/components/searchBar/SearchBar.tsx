import React, { useState } from "react";
import { View, TextInput } from "react-native";
import { styles } from "./SearchBarStyles";

interface SearchBarProps {
  updateTopic: (topic: string) => void;
}
const SearchBar: React.FC<SearchBarProps> = ({ updateTopic }) => {
  const [text, setText] = useState("");
  const onChange = (textValue: string) => setText(textValue);

  return (
    <View style={styles.container}>
      <TextInput
        placeholder="Search for news"
        style={styles.input}
        onChangeText={onChange}
        value={text}
        onSubmitEditing={() => {
          updateTopic(text);
          setText("");
        }}
      />
    </View>
  );
};
export default SearchBar;
