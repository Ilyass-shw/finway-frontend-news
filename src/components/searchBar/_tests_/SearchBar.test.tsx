import "@testing-library/jest-native/extend-expect";
import { render, fireEvent } from "@testing-library/react-native";
import SearchBar from "../SearchBar";

test("SearchBar should render", () => {
  const updateTopic = jest.fn() as (topic: string) => void;
  const { toJSON } = render(<SearchBar updateTopic={updateTopic} />);
  expect(toJSON()).toMatchSnapshot();
});

test("SearchBar should call uptadeTopic properly", () => {
  const updateTopic = jest.fn() as (topic: string) => void;
  const { getByPlaceholderText } = render(
    <SearchBar updateTopic={updateTopic} />
  );
  const input = getByPlaceholderText("Search for news");

  expect(updateTopic).toHaveBeenCalledTimes(0);
  fireEvent(input, "submitEditing");
  expect(updateTopic).toHaveBeenCalledTimes(1);
  expect(updateTopic).toHaveBeenCalledWith("");

  fireEvent.changeText(input, "some topic");
  fireEvent(input, "submitEditing");
  expect(updateTopic).toHaveBeenCalledTimes(2);
  expect(updateTopic).toHaveBeenCalledWith("some topic");
});
