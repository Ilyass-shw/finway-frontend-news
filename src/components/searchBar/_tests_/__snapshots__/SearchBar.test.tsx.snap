// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e6e6e6",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Search for news"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "fontSize": 15,
        "height": 60,
        "margin": 5,
        "marginVertical": 15,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
